### PR TITLE
Update `dependencies.yaml` Schema

### PR DESCRIPTION
**Pull request by ajschmidt8**
This PR updates the `dependencies.yaml` schema to help ensure compatibility with future output file types.

### CODE REVIEW

**Review by bdice — APPROVED** [review]
This schema change looks great to me.

**Comment by vyasr** [comment]
Same here, the schema changes look like what we discussed! Let us know if you run into snags updating the code to generate this.

File: README.md [hunk 1]
```diff
@@ -86,17 +86,17 @@ The result of the above configuration is that the following dependency files wou
...
 files:
   test:
-    generate: none
+    output: none
     includes:
       - test
 ```
 
-When `generate: none` is used, the `conda_dir`, `requirements_dir` and `matrix` keys can be ommitted. The use case for `generate: none` is described in the [_Additional CLI Notes_](#additional-cli-notes) section below.
+When `output: none` is used, the `conda_dir`, `requirements_dir` and `matrix` keys can be ommitted. The use case for `output: none` is described in the [_Additional CLI Notes_](#additional-cli-notes) section below.
```
Comment (bdice) [inline README.md:99]:
```suggestion
When `output: none` is used, the `conda_dir`, `requirements_dir` and `matrix` keys can be omitted. The use case for `output: none` is described in the [_Additional CLI Notes_](#additional-cli-notes) section below.
```

**Review by bdice — APPROVED** [review]
Quickly re-reviewed the new changes. Looks good.

**Pull request merged**